spec/requests/api/roles_spec.rb

```diff
@@ -0,0 +1,378 @@
+# - Assign a single feature               /api/roles/:id/features               POST, action "assign"
+# - Assign multiple features              /api/roles/:id/features               POST, action "assign"
+# - Un-assign a single feature            /api/roles/:id/features               POST, action "unassign"
+# - Un-assign multiple features           /api/roles/:id/features               POST, action "unassign"
+# - Delete a role                         /api/roles/:id                        DELETE
+# - Delete a role by action               /api/roles/:id                        action "delete"
+# - Delete multiple roles                 /api/roles                            action "delete"
+#
+require 'spec_helper'
+
+describe ApiController do
+  include Rack::Test::Methods
+
+  let(:feature_identifiers) do
+    %w(vm_explorer ems_infra_tag my_settings_time_profiles
+       miq_request_view miq_report_run storage_manager_show_list)
+  end
+  let(:expected_attributes) { %w(id name read_only settings) }
+  let(:sample_role1) do
+    {
+      "name"     => "sample_role_1",
+      "settings" => {"restrictions" => {"vms" => "user"}},
+      "features" => [
+        {:identifier => "vm_explorer"},
+        {:identifier => "ems_infra_tag"},
+        {:identifier => "my_settings_time_profiles"}
+      ]
+    }
+  end
+  let(:sample_role2) do
+    {
+      "name"     => "sample_role_2",
+      "settings" => {"restrictions" => {"vms" => "user_or_group"}},
+      "features" => [
+        {:identifier => "miq_request_view"},
+        {:identifier => "miq_report_run"},
+        {:identifier => "storage_manager_show_list"}
+      ]
+    }
+  end
+  let(:features_list) do
+    {
+      "features"  => [
+        {:identifier => "miq_request_view"},
+        {:identifier => "miq_report_run"},
+        {:identifier => "storage_manager_show_list"}
+      ]
+    }
+  end
+
+  before(:each) do
+    init_api_spec_env
+
+    @product_features = feature_identifiers.collect do |identifier|
+      FactoryGirl.create(:miq_product_feature, :identifier => identifier)
+    end
+  end
+
+  def app
+    Vmdb::Application
+  end
+
+  def test_features_query(role, role_url, klass, attr = :id)
+    api_basic_authorize
+
+    run_get role_url, :expand => "features"
+    expect_request_success
+
+    expect(@result).to have_key("name")
+    expect(@result["name"]).to eq(role.name)
+    expect(@result).to have_key("features")
+    expect(@result["features"].size).to eq(fetch_value(role.miq_product_features.count))
+
+    expect_result_resources_to_include_data("features", attr.to_s => klass.pluck(attr))
+  end
+
+  describe "Features" do
+    it "query available features" do
+      role = FactoryGirl.create(:miq_user_role,
+                                :name                 => "Test Role",
+                                :miq_product_features => @product_features)
+      url = roles_url + "/#{role.id}/"
+      test_features_query(role, url, MiqProductFeature, :identifier)
```

Comment:
No need for line 91,
this can simply be: 

test_features_query(role, roles_url(role.id), MiqProductFeature, :identifier)